### PR TITLE
Move Float16 print tests to their own file and test exhaustively

### PR DESCRIPTION
**Pull request by stephentyrone**
We can easily test all 2**16 values, so let's do it. Also now _Float16 is properly supported in clang, so we can pass arguments to CPP that way, which lets us get snan right on more platforms and just generally make things cleaner.

### CODE REVIEW

**Comment by stephentyrone** [comment]
@swift-ci test

File: test/stdlib/PrintFloat16.swift [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+//  // the C locale.
+//  PrintTests.setUp {
+//    if let localeArgIndex = CommandLine.arguments.firstIndex(of: "--locale") {
+//      let locale = CommandLine.arguments[localeArgIndex + 1]
+//      expectEqual("ru_RU.UTF-8", locale)
+//      setlocale(LC_ALL, locale)
+//    } else {
+//      setlocale(LC_ALL, "")
+//    }
+//  }
```
Comment (mikeash) [inline test/stdlib/PrintFloat16.swift:26]: Should this be uncommented, or perhaps deleted?

**Review by mikeash — APPROVED** [review]

**Comment by stephentyrone** [comment]
@swift-ci test

**Comment by stephentyrone** [comment]
@swift-ci smoke test

**Comment by stephentyrone** [comment]
@swift-ci smoke test

**Review by glessard — APPROVED** [review]

**Pull request merged**